src/main/java/main/internet/shop/model/User.java

```diff
@@ -6,16 +6,19 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 import main.internet.shop.model.role.UserRole;
+import main.internet.shop.utils.HashUtil;
 
 public class User {
     private Long id;
     private String name;
     private String login;
     private String password;
     private Set<UserRole> userRoles;
+    private byte[] salt;
 
     public User() {
         userRoles = new HashSet<>();
+        salt = HashUtil.generateSalt();
```

Comment:
this is part of the business logic
let's move the generation to UserServiceImpl class